### PR TITLE
user:delete command

### PR DESCRIPTION
**Pull request by pheinrichs**
Created php artisan user:delete command.
Used choice rather than table and bypass choice if not needed.
issue #63

### CODE REVIEW

File: app/Console/Commands/DeleteUser.php [hunk 1]
```diff
@@ -39,15 +39,22 @@ public function __construct()
...
-        $headers = ['ID','username','realname','email'];
         $user = $this->argument('user');
-        $user_list = User::select('user_id','username','realname','email')->where('username', 'like', '%'.$user.'%')->orWhere('realname','like', '%'.$user.'%')->get()->toArray();
-        $this->table($headers,$user_list);
-
-        $userid = $this->ask('Enter the ID you wish to remove');
-        $remove_user = User::find($userid);
-        if ($this->confirm('Do you wish to remove '.$remove_user->realname.'? [y|N]')) {
-            $remove_user->delete();
+        $user_list = User::select('username')->where('username', 'like', '%'.$user.'%')->orWhere('realname','like', '%'.$user.'%')->get();
```
Comment (pheinrichs) [inline app/Console/Commands/DeleteUser.php:67]: @murrant  I'm searching by realname and username just to make it a little easier to find accounts. Did we want to do this or just by username?

File: app/Console/Commands/DeleteUser.php [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        }
+        if (count($names) > 1) {
+            $name = $this->choice('Who would you like to remove?',$names, false);
+        }
+        else {
+            $name = $names[0];
+        }
+        if ($this->confirm('Do you wish to remove '.$name.'?')) {
+            $remove_user = User::select('user_id')->where('username', $name)->first();
+            User::find($remove_user->user_id)->delete();
```
Comment (murrant) [inline app/Console/Commands/DeleteUser.php]: You can just ->delete() the $remove_user object.

Comment (pheinrichs) [inline app/Console/Commands/DeleteUser.php]: Yeah, whoops. Not sure why I did that. Will fix.

**Comment by murrant** [comment]
I think it is fine to do a search based on username and realname.

Last thing to do is add tests.  If you want I can create a test for user:add to give you an idea.

**Comment by pheinrichs** [comment]
That would be great, I'd love an example to work off of.

**Comment by murrant** [comment]
It won't be until later today.

If you want to take a crack at it, just create a new file in /tests/commands and add tests there. There are other tests in the /tests folder to give an example.
You'll need to add a user then use either Artisan::call() or shell_exec() to test the command.  (not sure which is better without writing code :)

Run phpunit  or phpunit <test file> to run tests.

Hmm, we should probably have a little documentation on testing.

**Comment by pheinrichs** [comment]
I'll give it a go. See where i end up.

**Comment by murrant** [comment]
@pheinrichs A little later than I figured, but I went down a couple of wrong paths.

PR https://github.com/librenms/librenmsv2/pull/70

I merged this so you can merge it to your branch to avoid conflicts in phpunit.xml.

**Comment by pheinrichs** [comment]
@murrant Take a look when you get a chance I added a couple tests, see if there are any more you can think of.

File: tests/commands/UserDeleteTest.php [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        $mock->shouldReceive('confirm')
+            ->with('Do you wish to remove '.$username.'?')
+            ->once()
+            ->andReturn('yes');
+
+        $mock->shouldReceive('info')
+            ->once();
+
+        // call the handle function
+        $mock->handle();
```
Comment (murrant) [inline tests/commands/UserDeleteTest.php:61]: Add a check to make sure the user is deleted.

File: tests/commands/UserDeleteTest.php [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+            ->with('Do you wish to remove '.$removeUser.'?')
+            ->once()
+            ->andReturn('yes');
+
+        $mock->shouldReceive('info')
+            ->once();
+
+        // call the handle function
+        $mock->handle();
+    }
```
Comment (murrant) [inline tests/commands/UserDeleteTest.php]: Again add checks that one is deleted and the other is not.

File: tests/commands/UserDeleteTest.php [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+        $mock->shouldReceive('confirm')
+            ->with('Do you wish to remove '.$removeUser.'?')
+            ->once()
+            ->andReturn('yes');
+
+        $mock->shouldReceive('info')
+            ->once();
+
+        // call the handle function
+        $mock->handle();
```
Comment (murrant) [inline tests/commands/UserDeleteTest.php]: Again add checks that one is deleted and the other is not.

**Comment by murrant** [comment]
Looks great!  Just those two little additions and we'll merge it.

File: tests/commands/UserDeleteTest.php [hunk 6]
```diff
@@ -0,0 +1,93 @@
+<?php
+/*
+ * Copyright (C) 2016 Tony Murray <murraytony@gmail.com>
```
Comment (murrant) [inline tests/commands/UserDeleteTest.php]: Oops, this is your code :)

Comment (murrant) [inline tests/commands/UserDeleteTest.php]: Oops this is your code.

File: app/Console/Commands/DeleteUser.php [hunk 7]
```diff
@@ -0,0 +1,64 @@
+<?php
```
Comment (murrant) [inline app/Console/Commands/DeleteUser.php:1]: Add copyright for this file as well.

File: tests/commands/UserDeleteTest.php [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+
+        $mock->shouldReceive('confirm')
+            ->with('Do you wish to remove '.$removeUser.'?')
+            ->once()
+            ->andReturn('yes');
+
+        $mock->shouldReceive('info')
+            ->once();
+
+        // call the handle function
```
Comment (murrant) [inline tests/commands/UserDeleteTest.php:94]: Again add checks that one is deleted and the other is not.

File: tests/commands/UserDeleteTest.php [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * SettingsTest.php
```
Comment (murrant) [inline tests/commands/UserDeleteTest.php]: Wrong file name.

**Pull request merged**